app/services/record_declarations/base.rb

```diff
@@ -21,8 +21,8 @@ class Base
     delegate :schedule, :participant_declarations, to: :user_profile, allow_nil: true
 
     class << self
-      def call(params)
-        new(params).call
+      def call(params:)
+        new(params: params).call
```

Comment:
What is the benefit to have named parameter here?

Reply:
Consistency with the participants service.

And the benefit of that is... we can have shared examples between them to show the behaviour for "when user is not a participant" or "when user does not have access to the course".

It's the difference of `described_class.call(params)` vs `described_class.call(params: params)` in tests.

And, well, I opted for making it a named parameter rather than making participant actions one an un-named parameter because I thought that would lead to fewer changes.